tests/util/socket.py

```diff
@@ -0,0 +1,11 @@
+import socket
+
+
+def find_available_listen_port(name: str = "free") -> int:
+    s = socket.socket()
+    s.bind(("127.0.0.1", 0))
+    addr = s.getsockname()
+    assert addr[1] > 0
+    s.close()
+    print(f"{name} port: {addr[1]}")
+    return int(addr[1])
```

Comment:
technically, I believe it would be valid for an operating system to return the same port every time, since the port from the last call isn't necessarily busy by the time we call it again. However, it looks like they behave reasonably by randomizing a port.

If this turns out to be a problem, we can implement the randomization ourselves, but I'd rather keep things simple as long as it works.

Naming ports and printing them helps a lot in interpreting logs, I think it's good to keep it.